src/pages/SignUp/Field.tsx

```diff
@@ -0,0 +1,49 @@
+import Icon, { IconType } from "components/Icon";
+import { unpack } from "components/form/helpers";
+import { Classes } from "components/form/types";
+import { FieldValues, Path, get, useFormContext } from "react-hook-form";
+import { fieldClasses } from "./constants";
+
+type Props<T extends FieldValues> = {
+  classes?: Classes;
+  name: Path<T>;
+  placeholder: string;
+  icon?: IconType;
+};
+
+export default function Field<T extends FieldValues>(props: Props<T>) {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext<T>();
```

Comment:
we should steer away from baked in form context for new fields - as use case now doesn't need fields to be deep unlike that of crypto UIs before

Reply:
I'm not sure I understand, can you expand a bit on what you were trying to say?

Reply:
existing component `components/Field` has `useFormContext` inside and is often inside `FormProvider` - this made sense before as form inputs would often placed way deep in a complex UI (crypto admin UIs)

```js
<FormProvider>
     {....}
</FormProvider>
```
but new `SignUp/Field` could be abstracted in such a way that its api is just that of native `<input/>` and can be used as

```
//no need for FormProvider
<Field {...register("password")}/>
////no need for FormProvider
```
as [demonstrated here](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2718/files#diff-f176c879a94c9ef8e62a3aca7830390215dcb0cba4751b9f53019b3281a58fafR198)


new forms don't have to use `FormProvider`
no need for additional generics `Field<Additional>` as `useForm.register` already has autocompletion 

```js
<Field {...register("...field name selection...")}/>
```

Reply:
@ap-justin what do you think about renaming this component to `Input` and moving to */components/form/\** folder as part of this PR? Later on we can refactor it to use a slightly different logic as you described.
I'll need this component for my work to rework Login page and you'll probably need it too as part of your own work.